services/admin/src/admin/routes/obsolete_cache.py

```diff
@@ -0,0 +1,140 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import logging
+from typing import Optional, TypedDict
+
+from libcommon.dataset import get_supported_dataset_infos
+from libcommon.simple_cache import (
+    delete_dataset_responses,
+    get_all_datasets,
+    get_cache_count_for_dataset,
+)
+from libcommon.storage import StrPath
+from libcommon.viewer_utils.asset import delete_asset_dir
+from starlette.requests import Request
+from starlette.responses import Response
+
+from admin.authentication import auth_check
+from admin.utils import (
+    Endpoint,
+    UnexpectedError,
+    get_json_admin_error_response,
+    get_json_ok_response,
+)
+
+DATASET_CONFIG_NAMES_KIND = "dataset-config-names"
+MINIMUM_SUPPORTED_DATASETS = 20_000
+
+
+class DatasetCacheReport(TypedDict):
+    dataset: str
+    cache_records: int
+
+
+def get_supported_dataset_names(
+    hf_endpoint: str,
+    hf_token: Optional[str] = None,
+) -> set[str]:
+    supported_dataset_infos = get_supported_dataset_infos(hf_endpoint=hf_endpoint, hf_token=hf_token)
+    return {dataset_info.id for dataset_info in supported_dataset_infos}
+
+
+def get_obsolete_cache(
+    hf_endpoint: str,
+    hf_token: Optional[str] = None,
+) -> list[DatasetCacheReport]:
+    supported_dataset_names = get_supported_dataset_names(hf_endpoint=hf_endpoint, hf_token=hf_token)
+    existing_datasets = get_all_datasets(DATASET_CONFIG_NAMES_KIND)
```

Comment:
why do we filter on `kind="dataset-config-names"`? To get a faster result? We might miss some obsolete cache entries

Reply:
Yes it was because of performance. Do you think is better to run over the whole collection?

Reply:
I don't know. It depends on the difference in performance and in the results

Reply:
Done